src/internal/xml-parser.ts

```diff
@@ -394,3 +397,144 @@ export function parseBucketVersioningConfig(xml: string) {
+      msgCrcAccumulator = crc32(payLoadBuffer, msgCrcAccumulator)
+      // read the checksum early and detect any mismatch so we can avoid unnecessary further processing.
+      const messageCrcByteValue = Buffer.from(responseStream.read(4)).readInt32BE()
+      const calculatedCrc = msgCrcAccumulator.readInt32BE()
+      // Handle message CRC Error
+      if (messageCrcByteValue !== calculatedCrc) {
+        throw new Error(
+          `Message Checksum Mismatch, Message CRC of ${messageCrcByteValue} does not equal expected CRC of ${calculatedCrc}`,
+        )
+      }
+      payloadStream = readableStream(payLoadBuffer)
+    }
+    const messageType = headers['message-type']
+
+    switch (messageType) {
+      case 'error': {
+        const errorMessage = headers['error-code'] + ':"' + headers['error-message'] + '"'
+        throw new Error(errorMessage)
+      }
+      case 'event': {
+        const contentType = headers['content-type']
+        const eventType = headers['event-type']
+
+        switch (eventType) {
+          case 'End': {
+            selectResults.setResponse(res)
+            return selectResults
+          }
+
+          case 'Records': {
+            const readData = payloadStream?.read(payLoadLength)
+            selectResults.setRecords(readData)
+            break
+          }
+
+          case 'Progress':
+            {
+              switch (contentType) {
+                case 'text/xml': {
+                  const progressData = payloadStream?.read(payLoadLength)
+                  selectResults.setProgress(progressData.toString())
+                  break
+                }
+                default: {
+                  const errorMessage = `Unexpected content-type ${contentType} sent for event-type Progress`
+                  throw new Error(errorMessage)
+                }
+              }
+            }
+            break
+          case 'Stats':
+            {
+              switch (contentType) {
+                case 'text/xml': {
+                  const statsData = payloadStream?.read(payLoadLength)
+                  selectResults.setStats(statsData.toString())
+                  break
+                }
+                default: {
+                  const errorMessage = `Unexpected content-type ${contentType} sent for event-type Stats`
+                  throw new Error(errorMessage)
+                }
+              }
+            }
+            break
+          default: {
+            // Continuation message: Not sure if it is supported. did not find a reference or any message in response.
+            // It does not have a payload.
+            const warningMessage = `Un implemented event detected  ${messageType}.`
+            // eslint-disable-next-line no-console
+            console.warn(warningMessage)
+          }
+        } // eventType End
+      } // Event End
+    } // messageType End
+  } // Top Level Stream End
```

Comment:
Why do we even have to state end bracket comments? In every IDE or text editor, brackets are all collapsible. This is not a Dart code anyway.